File: Programmers/level_2/소수 찾기/solution.py
from itertools import permutations
import math

def solution(numbers):
    answer = 0
    
    permutedArray = []
    
    for i in range(1, len(numbers) + 1):
        tempArray = list(permutations(numbers,i))
        
        for item in tempArray:
            permutedArray.append(''.join(item))
    
    permutedArray = list(set(list(map(int,permutedArray))))
    
    for p in permutedArray:
        isAnswer = 1
        if p == 1 or p == 0 :
            continue
            
        for i in range(2, p):
            if p % i == 0 :
                isAnswer = 0
                break
        
        answer += isAnswer
            
            
    
    return answer

File: Programmers/level_2/소수 찾기/solution.swift
import Foundation

func solution(_ numbers: String) -> Int {
    
    var numberArray = numbers.map { String($0) }
    let length = numberArray.count
    var visited = Array(repeating: false, count: length)
    var tempArray = [String]()
    var permuted = Set<Int>()
    var answer = 0
    
    func permutations(_ count: Int) {
        if tempArray.count == count {
            let number = Int(tempArray.joined(separator: ""))!
            permuted.insert(number)
            return
        }
        
        for i in 0 ..< length {
            if visited[i] {
                continue
            }
            
            visited[i] = true
            tempArray.append(numberArray[i])
            
            permutations(count)
            
            visited[i] = false
            tempArray.removeLast()
        }
    }
    
    for i in 1 ... length {
        permutations(i)
    }
    
    for number in Array(permuted) {
        if isPrime(number) {
            answer += 1
        }
    }
    
    
    return answer
}

func isPrime(_ number: Int) -> Bool {
    if number <= 1 {
        return false
    } else if number <= 3  {
        return true
    }
    
    let num = Int(sqrt(Double(number)))

    for i in 2 ... num{
        if number % i == 0 {
            return false
        }
    }
    
    return true
}